SI-6886: Deprecate -expand-all-types

janekdb opened the pull request:
This Scaladoc flag was added to assist with debugging and was not
intended to have general availability.

If an element requires a separate Scaladoc page add @documentable.

Discussion leading to deprecation decision is here: https://groups.google.com/forum/?fromgroups=#!topic/scala-user/GA8yIvNuKkg

janekdb commented:
Review by @VladUreche, @mpociecha .

VladUreche commented:
I'm not sure this is an acceptable deprecation roadmap. Normally a feature planned for removal must be available for an entire minor (2.12) with the deprecation message. Why is it so urgent to remove this flag?

Ping @SethTisue, WDYT?

VladUreche commented:
On the other hand, I agree this is an obscure and confusing flag, so from my perspective, it's perfectly fine to remove it if there's a blessing to shortcut the normal deprecation procedures.

som-snytt commented:
2.13 is the Deprecocalypse, according to soc.

janekdb committed:
SI-6886: Deprecate -expand-all-types

This Scaladoc flag was added to assist with debugging and was not
intended to have general availability.

If an element requires a separate Scaladoc page add @documentable.

janekdb commented:
There is no urgency to remove this. I wasn't aware of the full deprecation policy.

No smoke was seen when testing,

    ✔ ~/workspaces/public/scala [2.11.x-deprecate-expand-all-types ↑·1|✚ 1…7⚑ 23] 
    20:27 $ echo "object K" > /tmp/K.scala && build/quick/bin/scaladoc -expand-all-types -d /tmp/ /tmp/K.scala 
    warning: there was one deprecation warning; re-run with -deprecation for details
    model contains 2 documentable templates
    one warning found

    ✔ ~/workspaces/public/scala [2.11.x-deprecate-expand-all-types ↑·1|✚ 1…7⚑ 23] 
    20:28 $ echo "object K" > /tmp/K.scala && build/quick/bin/scaladoc -expand-all-types -deprecation -d /tmp/ /tmp/K.scala 
    warning: -expand-all-types is deprecated: This flag is scheduled for removal in 2.13. If you have a case where you need this flag then please report a bug.
    model contains 2 documentable templates
    one warning found

VladUreche commented:
> 2.13 is the Deprecocalypse, according to soc.

Yes, the final (Type) Judgment day.

SethTisue commented:
I'll take a look at this after 2.11.8 is out the door.

SethTisue commented:
I think this should target 2.12. (Sorry, Janek, for the busywork for retargeting.)

janekdb commented:
@SethTisue - Always more than happy to sling a few `git` commands around to get to the right place!

Replaced with #5010 which is a new branch with this cherry-picked to. If there's a better way to move commits between branches on the same PR feel free to advise - I'll be able to try it out on the other PRs that need to migrate to `2.12.x`.

SethTisue commented:
> If there's a better way to move commits between branches on the same PR feel free to advise

sadly no, a GitHub limitation.

som-snytt commented:
> If there's a better way to move commits between branches on the same PR feel free to advise

As @SethTisue seems to know instinctively, the better way is to advise someone else to do it.

That sounds meaner than intended, but it's too funny to delete outright. Too much irony? Or irony in both clauses?

As Seth has learned from his time at Typebend swabbing our PRs,...

janekdb commented:
Seems I've joined a crowd of > 1300 others who would also like to see this feature in GitHub: https://github.com/isaacs/github/issues/18 !
